File: src/main/scala/com/kapilkoju/autopos/party/domain/Vehicle.scala
package com.kapilkoju.autopos.party.domain

import javax.persistence._
import javax.validation.constraints.{NotNull, Size}

import com.kapilkoju.autopos.kernel.domain.AuditableEntity

import scala.beans.BeanProperty


@Entity
@Table(name = "vehicle")
class Vehicle extends AuditableEntity {

  @NotNull
  @Size(min = 1, max = 20)
  @Column(name = "number", length = 20, nullable = false, unique = true)
  @BeanProperty
  var number: String = _

  @Size(max = 250)
  @Column(name = "remarks", length = 250)
  @BeanProperty
  var remarks: String = _

  @ManyToOne
  @BeanProperty
  var owner: Customer = _

  def getTitle: String = {
    if (owner == null) {
      number
    } else {
      s"[ $number ] ${owner.getName}"
    }
  }
}

File: src/main/kotlin/com/kapilkoju/autopos/party/domain/Vehicle.kt
package com.kapilkoju.autopos.party.domain


import com.kapilkoju.autopos.kernel.domain.AuditableEntity

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.persistence.Table
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
@Table(name = "vehicle")
data class Vehicle(

        @NotNull
        @Size(min = 1, max = 20)
        @Column(name = "number", length = 20, nullable = false, unique = true)
        val number: String,

        @Size(max = 250)
        @Column(name = "remarks", length = 250)
        val remarks: String?,

        @ManyToOne
        val owner: Customer?

) : AuditableEntity() {

    fun getTitle(): String {
        return if (owner == null) {
            number
        } else {
            "[ $number ] ${owner.name}"
        }
    }


}
